packages/shared/src/components/profile/ProfileTooltip.tsx

```diff
@@ -38,9 +41,29 @@ export function ProfileTooltip({
     userId: user.id,
     requestUserInfo: true,
   });
+  const query = useQueryClient();
+  const handler = useRef<() => void>();
+
+  const onShow = () => {
+    if (!scrollingContainer) {
+      return;
+    }
+
+    scrollingContainer.removeEventListener('scroll', handler.current);
+    handler.current = () => query.setQueryData(['readingRank', user.id], {});
+    scrollingContainer.addEventListener('scroll', handler.current);
```

Comment:
I'm not a 100% sure why we need to manually set the queryData here on the scroll listener?
Is a regular reference to the element not enough?

Reply:
This is to clear the current tooltip and close it after scrolling - we force to empty the data so it would be unmounted.

> Is a regular reference to the element not enough?
That's actually a great point, was trying to make it abide with React's approach. Should we move to native and just pass a selector? What do you think?

Reply:
I think it might be good enough with regular reference as this might be a bit overkill,  but leave it up to you if this is needed for some other reason then it can make sense.

Reply:
I think for the moment, we can move this now. Once we find other areas requiring this implementation, I think that can be worth moving this existing implementation to query selector.